backend/inspirehep/records/api/literature.py

```diff
@@ -335,9 +335,14 @@ def import_article(identifier):
     ).one_or_none()
 
     if pid:
+        pid_for_recid = PersistentIdentifier.query.filter_by(
+            pid_provider="recid", object_uuid=pid.object_uuid
+        ).one_or_none()
```

Comment:
* you need to add `object_type='rec'` to be able to use an index (see https://github.com/inveniosoftware/invenio-pidstore/blob/master/invenio_pidstore/models.py#L98).
* the proper way to do this would be with a self-join on the pidstore in order to do one fewer query. You need to alias the table for that: https://docs.sqlalchemy.org/en/13/orm/tutorial.html#using-aliases (we can discuss this).